keep-header: add page (#8401)
tldr-pages/tldr: pages/common/keep-header.md

# keep-header

> > Convenience utility that runs Unix commands in a header-aware fashion.

Reviewer: What does this do? After reading all of the command examples, I still have no idea.

From what I gather, it doesn't pipe the first line of a file/text to a command

Author: It pipes line 1 directly to stdout

Line 2 to EOF are passed to command

Basically, line 1 is short-circuited to stdout

Does this make sense now?

Reviewer: I think I now understand, it just allows me to make adequate suggestions, thanks for the help.

> > Convenience utility that runs Unix commands in a header-aware fashion.

Reviewer: ```suggestion
> Ignore the first line of a file when running a command.
```

Does this describe the command in an easier way in your opinion?

Author: What about:
`Keep first line untouched by a command, passing it directly to stdout`

Ignoring does not specify that the line ends up in stdout - could just not be passed through?

> More information: <https://github.com/eBay/tsv-utils#keep-header>.

- Sort a file and keep the first line at the top:

`keep-header {{path/to/file}} -- sort`

> - Output first line directly to stdout, pass rest through command:

Reviewer: ```suggestion
- Output first line directly to stdout, passing the remainder of the file through the specified command command:
```

> - Sort files except for retaining a single header line:

Reviewer: How is it sorting?
alphabetically, line by line?

Author: Well it sorts using `sort` so the general `sort` behaviour applies. Does that make sense?

Reviewer: :facepalm: completely forgot about the sort command, I thought it a feature of the command.

Author: No worries, that's probably down to me not explaining the purpose well!

`keep-header {{path/to/file}} -- sort`

- Read from stdin, sorting all except the first line:

> 

Reviewer: is the header the same as the first line?

Author: Yes it's two names for the same thing

`cat {{path/to/file}} | keep-header -- {{command}}`

> - Grep a file, keeping the first line whether it matches or not:

Reviewer: ```suggestion
- Grep a file, keeping the first line regardless of the search pattern:
```
